### PR TITLE
Update min cli core version from 2.30.0 to 2.38.0 to fix the compatibility issue

### PR DESCRIPTION
**Pull request by jiec-msft**
update min cli core version from 2.30.0 to 2.38.0 to fix the compatibility issue

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az spring

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring/HISTORY.md [hunk 1]
```diff
@@ -1,13 +1,11 @@
 Release History
 ===============
 
-1.1.7
+1.1.8
```
Comment (jiec-msft) [inline src/spring/HISTORY.md:4]: We'll skip `1.1.6` and `1.1.7` on purpose

Comment (zhoxing-ms) [inline src/spring/HISTORY.md:4]: Why not use version `1.1.7` but skip it?

**Comment by yonzhan** [comment]
spring

File: src/spring/HISTORY.md (hunk 1 above)
Comment (zhoxing-ms) [reply src/spring/HISTORY.md:4]: But we only hide version `1.1.6` at present, so the version `1.1.7` has not been released, right?

Comment (jiec-msft) [reply src/spring/HISTORY.md]: Oh, got catch! Just released that feature owner updated `1.1.7` in history but not `setup.py`. Yes, we can use `1.1.7`. Updated!

**Comment by jiec-msft** [comment]
@zhoxing-ms 

Do not merge.

This PR was created to fix an urgent issue. But now, the issue was mitigated with PR: https://github.com/Azure/azure-cli-extensions/pull/5360, we'll get back to this PR on next week.

Thank you!

File: src/spring/HISTORY.md (hunk 1 above)
Comment (jiec-msft) [reply src/spring/HISTORY.md]: Yes, 
1. `1.1.7` hasn't been updated into `setup.py`.
2. Also `1.1.7` hasn't been merged into `index.json`.

**Comment by jiec-msft** [comment]
Add some test result from Azure Spring Apps team:

1. CLI core 2.38.0 with spring extension 1.1.7
Azure DevOps(ADO) Link: https://msazure.visualstudio.com/AzureDMSS/_build/results?buildId=60609530&view=results
Screenshot:
image

The failed case is an unknown issue, and will be fixed later.

2. CLI core 2.39.0 with spring extension 1.1.7
ADO Link: https://msazure.visualstudio.com/AzureDMSS/_build/results?buildId=60609522&view=results
Screenshot:
image
The failed case is an unknown issue, and will be fixed later.


3. CLI core 2.40.0 with spring extension 1.1.7
ADO Link: https://msazure.visualstudio.com/AzureDMSS/_build/results?buildId=60609513&view=results
Screenshot:
image

The failed case is an unknown issue, and will be fixed later.


Take 2.38.0 test release to verify test environment:
image

**Comment by jiec-msft** [comment]
Add evidence that Azure CLI Core `2.37.0` is not compatible with our extension spring V1.1.6

image

**Comment by jiec-msft** [comment]
To add some more context of Azure CLI change that breaks the compatibility.
https://github.com/Azure/azure-cli/pull/22676

**Comment by jiec-msft** [comment]
@zhoxing-ms Our extension spring 1.1.7 has passed internal tests, now it's ready to be reviewed, merged, and released.

**Review by zhoxing-ms — APPROVED** [review]

**Comment by jiec-msft** [comment]
Link the feature request: https://github.com/Azure/azure-cli-extensions/issues/4640

**Pull request merged**